### PR TITLE
add the jit_check_app_addr_and_modify

### PR DESCRIPTION
**Pull request by septa2112**

### CODE REVIEW

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 1]
```diff
@@ -144,6 +144,105 @@ post_return(JitCompContext *cc, const WASMType *func_type, JitReg first_res)
...
+    WASMType *func_type;
+    JitFrame *jit_frame = cc->jit_frame;
+    JitReg native_ret;
+    JitReg fast_jit_func_ptrs, jitted_code = 0;
+    uint32 jitted_func_idx;
+
+    /* new var */
+    const char *signature = NULL;
+
+    if (func_idx >= wasm_module->import_function_count) {
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:162]: How about moving these codes (L163 to L168) to the next else branch (after L195)?

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 2]
```diff
@@ -144,6 +144,103 @@ post_return(JitCompContext *cc, const WASMType *func_type, JitReg first_res)
...
+                 NEW_CONST(I32, (uint32)sizeof(void *) * jitted_func_idx));
+    }
+
+    if (func_idx < wasm_module->import_function_count) {
+        func_import = &wasm_module->import_functions[func_idx].u.function;
+        func_type = func_import->func_type;
+        signature = func_import->signature;
+
+        JitReg import_func_ptrs = get_import_func_ptrs_reg(jit_frame);
+        JitReg native_func = jit_cc_new_reg_ptr(cc);
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:177]: Define the variables at the beginning of a scope, e.g. after L171.

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 3]
```diff
@@ -144,6 +144,103 @@ post_return(JitCompContext *cc, const WASMType *func_type, JitReg first_res)
...
+
+        GEN_INSN(LDPTR, native_func, import_func_ptrs,
+                 NEW_CONST(I32, (uint32)sizeof(void *) * func_idx));
+
+        for (int i = 0; i < func_type->param_count; i++) {
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:182]: Same as above, define `int i` at the beginning of L171

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 4]
```diff
@@ -144,6 +144,103 @@ post_return(JitCompContext *cc, const WASMType *func_type, JitReg first_res)
...
+                }
+            }
+        }
+    }
+    else {
+        func = wasm_module
+                   ->functions[func_idx - wasm_module->import_function_count];
+        func_type = func->func_type;
+
+        JitReg res = 0;
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:198]: Same as above

Comment (wenyongh) [reply core/iwasm/fast-jit/fe/jit_emit_function.c:198]: Should call pre_call in this branch.

File: core/iwasm/fast-jit/fe/jit_emit_function.c (hunk 1 above)
Comment (septa2112) [reply core/iwasm/fast-jit/fe/jit_emit_function.c:162]: OK, I will check the code.

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 5]
```diff
@@ -157,29 +157,81 @@ check_app_addr_and_convert_draft(JitCompContext *cc, WASMFunction *func_ctx,
...
+pre_load(JitCompContext *cc, JitReg *argvs, const WASMType *func_type)
+{
+    JitReg value;
+    uint32 i, outs_off;
+    // JitReg argv;
+
+    /* Prepare parameters for the function to call */
+    outs_off =
+        cc->total_frame_size + offsetof(WASMInterpFrame, lp)
+        + wasm_get_cell_num(func_type->types, func_type->param_count) * 4;
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:170]: No need to create outs_off and store reg into it, just store the popped reg into argvs[].

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 6]
```diff
@@ -157,29 +157,81 @@ check_app_addr_and_convert_draft(JitCompContext *cc, WASMFunction *func_ctx,
...
+
+    for (i = 0; i < func_type->param_count; i++) {
+        switch (func_type->types[func_type->param_count - 1 - i]) {
+            case VALUE_TYPE_I32:
+                value = jit_cc_new_reg_I32(cc);
+                PUSH_I32(value);
+                outs_off -= 4;
+                GEN_INSN(STI32, *argvs, value, NEW_CONST(I32, outs_off));
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:178]: No need to store, should be:
```C
POP_I32(value);
argv[func_type->param_count - 1 - i] = value;
```

File: core/iwasm/fast-jit/fe/jit_emit_function.c [hunk 7]
```diff
@@ -188,61 +240,34 @@ jit_compile_op_call_draft(JitCompContext *cc, uint32 func_idx, bool tail_call)
...
                  NEW_CONST(I32, (uint32)sizeof(void *) * func_idx));
 
-        /* Initialize parameter types */
-        total_size = sizeof(uint8) * (uint64)(func_type->param_count + 1);
+        /* allocate memory for argvs*/
+        total_size = sizeof(JitReg) * (uint64)(func_type->param_count);
         if (total_size >= UINT32_MAX
-            || !(param_types = wasm_runtime_malloc((uint32)total_size))) {
-            /* exception */
+            || !(argvs = wasm_runtime_malloc((uint32)total_size))) {
```
Comment (wenyongh) [inline core/iwasm/fast-jit/fe/jit_emit_function.c:246]: Use jit_malloc or jit_calloc, in fast-jit we usually use them.

Comment (wenyongh) [reply core/iwasm/fast-jit/fe/jit_emit_function.c:246]: Should jit_free(argvs) before function return: before return true or return false.

**Pull request closed without merging**